external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -128,27 +128,63 @@ const MANIFEST_FILE_NAME: &str = "Move.toml";
 const STD_LIB_PKG_ADDRESS: &str = "0x1";
 type SourceFiles = BTreeMap<FileHash, (FileName, String, bool)>;
 
+/// Information about compiled program (ASTs at different levels)
+#[derive(Clone)]
+struct CompiledProgram {
+    parsed: P::Program,
+    typed: T::Program,
+}
+
+/// Information about cached dependencies used during compilation and analysis
+#[derive(Clone)]
+struct CachedDeps {
+    /// Cached fully compiled program representing dependencies
+    compiled_program: Arc<FullyCompiledProgram>,
+    /// Cached symbols computation data for dependencies
+    symbols_data: Option<Arc<SymbolsComputationData>>,
+}
+
 /// Information about the compiled package and data structures
-/// computed during compilation
+/// computed during compilation and analysis
 #[derive(Clone)]
 pub struct CompiledPkgInfo {
-    parsed_program: P::Program,
-    typed_program: T::Program,
-    libs: Option<Arc<FullyCompiledProgram>>,
+    /// Package path
+    path: PathBuf,
+    /// Manifest hash
+    manifest_hash: Option<FileHash>,
```

Comment:
Is the manifest file the "right" thing to do here? It is possible a downstream dependency lives locally and got changed, and it appears this would not pick that change up. Is that correct?

Reply:
You are right, if the local dependency changes (whether direct or indirect), the IDE will not know about it. The idea here is that indeed if dependencies are changed "in the background" during a single editing session, things may indeed go out of sync, in which case re-starting the IDE will fix it (as everything will be re-fetched from disk).

In addition to what you observed, we also do not actually re-fetch (direct) remote dependencies if they change in the remote repo (GitHub). This is because the check for remote change had to happen at each compilation and was slowing down compilation/analysis enough that things like auto-completion was being too slow. Since we already do not track that, it seemed like only tracking explicit dependency changes (via manifest file changes) makes sense, as there is already a way when dependencies can change during a single editing session.

Reply:
Another way to look at it is that we mostly care about actions that user takes within the IDE (e.g., modify the manifest file), and largely do not care for actions taken outside of IDE (e.g., silent update of dependencies)